openedx/core/djangoapps/user_api/accounts/utils.py

```diff
@@ -176,3 +178,17 @@ def retrieve_last_sitewide_block_completed(username):
         course_key=text_type(item.location.course_key),
         location=text_type(item.location),
     )
+
+
+def generate_password(length=12, chars=string.letters + string.digits):
+    """Generate a valid random password"""
+    if length < 8:
+        raise ValueError("password must be at least 8 characters")
+
+    choice = random.SystemRandom().choice
+
+    password = ''
+    password += choice(string.digits)
+    password += choice(string.letters)
+    password += ''.join([choice(chars) for _i in xrange(length - 2)])
+    return password
```

Comment:
yes, it would need to be updated if there are other requirements on other systems.  But this is not solving for that case which already exists.

This PR consolidates random generation into a single method so it's more clear what to edit to create valid passwords.